linera-core/src/unit_tests/worker_tests.rs

```diff
@@ -79,294 +81,349 @@ use crate::{
+            application_permissions: Default::default(),
+        };
+        let admin_description = ChainDescription::new(origin, config, Timestamp::from(0));
+        storage
+            .write_blob(&Blob::new_chain_description(&admin_description))
+            .await
+            .expect("writing a blob should not fail");
+
+        let worker = WorkerState::new(
+            "Single validator node".to_string(),
+            Some(validator_keypair.secret_key),
+            storage,
+            NonZeroUsize::new(10).expect("Chain worker limit should not be zero"),
+        )
+        .with_allow_inactive_chains(is_client)
+        .with_allow_messages_from_deprecated_epochs(is_client)
+        .with_long_lived_services(has_long_lived_services)
+        .with_grace_period(Duration::from_micros(TEST_GRACE_PERIOD_MICROS));
+        Self {
+            committee,
+            worker,
+            admin_description,
+            admin_keypair: account_secret,
+            other_chains: BTreeMap::new(),
+        }
+    }
+
+    fn admin_id(&self) -> ChainId {
+        self.admin_description.id()
+    }
+
+    fn committee(&self) -> &Committee {
+        &self.committee
+    }
+
+    fn worker(&self) -> &WorkerState<S> {
+        &self.worker
+    }
+
+    fn admin_public_key(&self) -> AccountPublicKey {
+        self.admin_keypair.public()
+    }
+
+    async fn add_root_chain(
+        &mut self,
+        index: u32,
+        owner: AccountOwner,
+        balance: Amount,
+    ) -> ChainDescription {
+        let origin = ChainOrigin::Root(index);
+        let config = InitialChainConfig {
+            admin_id: Some(self.admin_id()),
+            epoch: self.admin_description.config().epoch,
+            ownership: ChainOwnership::single(owner),
+            committees: self.admin_description.config().committees.clone(),
+            balance,
+            application_permissions: Default::default(),
+        };
+        let description = ChainDescription::new(origin, config, Timestamp::from(0));
+        self.other_chains
+            .insert(description.id(), description.clone());
+        self.worker
             .storage
-            .create_chain(
-                committee.clone(),
-                ChainId::root(0),
-                description,
-                owner,
-                balance,
-                Timestamp::from(0),
-            )
+            .create_chain(description.clone())
             .await
             .unwrap();
+        description
     }
-    (committee, worker)
-}
 
-/// Same as `init_worker` but also instantiate a single initial chain.
-async fn init_worker_with_chain<S>(
-    storage: S,
-    description: ChainDescription,
-    owner: AccountOwner,
-    balance: Amount,
-) -> (Committee, WorkerState<S>)
-where
-    S: Storage + Clone + Send + Sync + 'static,
-{
-    init_worker_with_chains(storage, [(description, owner, balance)]).await
-}
+    async fn add_child_chain(
```

Comment:
Do we ever actually do that? Insert a child chain just deus ex machina, without an `OpenChain` operation?